File: conformance-tests/complex/graphql-parser-parse-variable-definition_m5.ts
class Parser {
  parseVariableDefinition(): VariableDefinitionNode {
      return this.node<VariableDefinitionNode>(this._lexer.token, {
        kind: Kind.VARIABLE_DEFINITION,
        variable: this.parseVariable(),
        type: (this.expectToken(TokenKind.COLON), this.parseTypeReference()),
        defaultValue: this.expectOptionalToken(TokenKind.EQUALS)
          ? this.parseConstValueLiteral()
          : undefined,
        directives: this.parseConstDirectives(),
      });
    }

  }

File: conformance-tests/complex/graphql-parser-parse-variable-definition_m5.lua
local Packages --[[ ROBLOX comment: must define Packages module ]]
local LuauPolyfill = require(Packages.LuauPolyfill)
local Boolean = LuauPolyfill.Boolean
type Parser = { parseVariableDefinition: (self: Parser) -> VariableDefinitionNode }
type Parser_statics = { new: () -> Parser }
local Parser = {} :: Parser & Parser_statics;
(Parser :: any).__index = Parser
function Parser.new(): Parser
	local self = setmetatable({}, Parser)
	return (self :: any) :: Parser
end
function Parser:parseVariableDefinition(): VariableDefinitionNode
	local refArg0 = self._lexer.token
	local refProp0 = Kind.VARIABLE_DEFINITION
	local refProp1 = self:parseVariable()
	self:expectToken(TokenKind.COLON)
	local refProp2 = self:parseTypeReference()
	return self:node(refArg0, {
		kind = refProp0,
		variable = refProp1,
		type = refProp2,
		defaultValue = if Boolean.toJSBoolean(self:expectOptionalToken(TokenKind.EQUALS))
			then self:parseConstValueLiteral()
			else nil,
		directives = self:parseConstDirectives(),
	})
end
